core/src/dispatch/mod.rs

```diff
@@ -202,6 +209,58 @@ impl Default for NetworkConfig {
     }
 }
 
+/// A port providing `NetworkStatusUpdates´ to listeners.
+pub struct NetworkStatusPort;
+impl Port for NetworkStatusPort {
+    type Indication = NetworkStatus;
+    type Request = NetworkStatusRequest;
+}
+
+/// Information regarding changes to the systems connections to remote systems
+#[derive(Clone, Debug)]
+pub enum NetworkStatus {
+    /// Indicates that a connection has been established to the remote system
+    ConnectionEstablished(SystemPath),
+    /// Indicates that a connection has been lost to the remote system.
+    /// The system will automatically try to recover the connection for a configurable amount of
+    /// retries. The end of the automatic retries is signalled by a `ConnectionDropped` message.
+    ConnectionLost(SystemPath),
```

Comment:
If we are going to implement session ids #84 we should put them here, too, so receivers know which session this is affecting.